var/spack/repos/builtin/packages/py-ucsf-pyem/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyUcsfPyem(PythonPackage):
+    """UCSF pyem is a collection of Python modules and command-line
+    utilities for electron microscopy of biological samples."""
+
+    homepage = "https://github.com/asarnow/pyem"
+    git      = "https://github.com/asarnow/pyem.git"
+
+    maintainers = ['dorton21']
+
+    # Using commit since releases haven't been updated in 2 years
+    version('2021-04-07', commit='ed0527f98657d21d887357426b74e5240d477fae')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-future@0.15:', type=('build', 'run'))
+    depends_on('py-numba@0.41:', type=('build', 'run'))
+    depends_on('py-seaborn@0.9:', type=('build', 'run'))
+    depends_on('py-pandas@0.23.4:', type=('build', 'run'))
+    depends_on('py-pathos@0.2.1:', type=('build', 'run'))
+    depends_on('py-pyfftw@0.10:', type=('build', 'run'))
+    depends_on('py-healpy@1.11:', type=('build', 'run'))
+    depends_on('py-natsort@6.0:', type=('build', 'run'))
+
+    # Extracts files into bin for proper useage
+    @run_after('install')
+    def extraction(self):
+        install_tree('.', self.prefix.bin)
```

Comment:
Doesn't this also install all source code to bin? I don't think we want that.

Reply:
If I'm not mistaken this package is simply used via calling scripts like `star.py --copy-micrograph-coordinates original_particles.star from_cryosparc.star from_csparc_with_coords.star
` as seen here: https://github.com/asarnow/pyem/wiki/Export-from-cryoSPARC-v2
The default install doesn't give access to the scripts ergo a simple dump seems to work. Unless I'm misunderstanding how this package works altogether.

Reply:
What does the `python setup.py build` and `python setup.py install` steps do then? I wish they used entrypoints like normal Python packages os they would automatically get installed in `prefix.bin`.

Reply:
The default installation does not create a bin. The only files paths that are created are without the install_tree() dump are:
`py-ucsf-pyem-2021-04-07-.../lib/python3.8/site-packages/pyem/...` which does not contain all the correct scripts. And `py-ucsf-pyem-2021-04-07-.../lib/python3.8/site-packages/pyem-0.4-py3.8.egg-info/...` is also created which does not contain anything useful. Also a perl5 folder which is empty. I may be mistaken but I think the setup.py basically checks the required packages exist and that's about it.

Reply:
Would it be sufficient to do:
```suggestion
        mkdir(self.prefix.bin)
        install('*.py', self.prefix.bin)
```
or are there other files that are needed?

Reply:
After some testing, it looks like this suggestion will work. Good idea.